counties/counties.go

```diff
@@ -0,0 +1,50 @@
+package counties
+
+import (
+	"fmt"
+
+	"github.com/corbaltcode/usps/zip4"
+)
+
+func CollectZip4Details(tarName, zipPassword string) (map[string][]string, error) {
+	var details []zip4.Zip4Detail
+
+	yield := func(detail zip4.Zip4Detail) {
+		details = append(details, detail)
+	}
+
+	err := zip4.ReadZip4FromZip4Tar(tarName, zipPassword, yield)
+	if err != nil {
+		return nil, fmt.Errorf("error processing ZIP+4 data: %v", err)
+	}
+
+	return mapZipToCounty(details), nil
+}
+
+func mapZipToCounty(details []zip4.Zip4Detail) map[string][]string {
+	zipToCounty := make(map[string][]string)
+
+	for _, detail := range details {
+		zip := detail.ZipCode
+		county := detail.CountyNumber
+
+		// Check if the county number is already in the list for the zip.
+		counties, exists := zipToCounty[zip]
```

Comment:
`zipToCounty[zip]` returns a nil slice (the zero value of `[]string`) if `zip` doesn't exist in the map, and nil slices behaves like any zero-length slice. So I don't think you need the `exists` here or the `if` below.

Reply:
Ah, gotcha. I'm still getting used some of go's nuances. I've updated the logic to eliminate the redundant check. When time allows, please take a look and let me know what you think of the updated organization.